hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/rollout/RolloutScheduler.java

```diff
@@ -43,14 +48,56 @@ public RolloutScheduler(
     @Scheduled(initialDelayString = PROP_SCHEDULER_DELAY_PLACEHOLDER, fixedDelayString = PROP_SCHEDULER_DELAY_PLACEHOLDER)
     public void runningRolloutScheduler() {
         log.debug("rollout schedule checker has been triggered.");
-
-        // run this code in system code privileged to have the necessary permission to query and create entities.
+        // run this code in system code privileged to have the necessary
+        // permission to query and create entities.
         systemSecurityContext.runAsSystem(() -> {
-            // workaround eclipselink that is currently not possible to execute a query without multi-tenancy if MultiTenant
-            // annotation is used. https://bugs.eclipse.org/bugs/show_bug.cgi?id=355458. So, iterate through all tenants and execute the rollout
-            // check for each tenant separately.
-            systemManagement.forEachTenant(tenant -> rolloutHandler.handleAll());
+            // workaround eclipselink that is currently not possible to
+            // execute a query without multi-tenancy if MultiTenant
+            // annotation is used.
+            // https://bugs.eclipse.org/bugs/show_bug.cgi?id=355458. So
+            // iterate through all tenants and execute the rollout check for
+            // each tenant seperately.
+
+            systemManagement.forEachTenant(this::handle);
             return null;
         });
     }
+
+    private void handle(String tenant) {
+        log.trace("Handling rollout for tenant: {}", tenant);
+        if (rolloutTaskExecutor == null) {
+            rolloutHandler.handleAll();
+        } else {
+            handleParallel(tenant);
+        }
+    }
+
+    private void handleParallel(String tenant) {
+        rolloutTaskExecutor.submit(() -> systemSecurityContext.runAsSystemAsTenant(() -> {
+            try {
+                rolloutHandler.handleAll();
+            } catch (Exception e) {
+                log.error("Error processing rollout for tenant {}", tenant, e);
```

Comment:
this catch shall also be present in non-parallel case?
maybe it will be better to have handle method
```java
log.trace("Handling rollout for tenant: {}", tenant);
try {
    rolloutHandler.handleAll();
} catch (Exception e) {
    log.error("Error processing rollout for tenant {}", tenant, e);
}
```
which shall be called directly or in "rolloutTaskExecutor.submit(() -> systemSecurityContext.runAsSystemAsTenant()"